docs/proposals/choosing-scheduler.md

```diff
@@ -0,0 +1,165 @@
+
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+<strong>
+The latest release of this document can be found
+[here](http://releases.k8s.io/release-1.1/docs/proposals/choosing-scheduler.md).
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# Choosing scheduler in a multi-scheduler configuration
+
+## Introduction
+
+This document proposes a mechanism for assigning a scheduler to a pod
+in a system which is using multiple schedulers as described [here](multiple-schedulers.md).
+In particular, we decribe an admission controller which adds the scheduler name annotation,
+and which takes its configuration from a new API object called `PodPolicy` that determines,
+for each pod, which scheduler is responsible for scheduling that pod, based on various
+characteristics of the pod.
+
+## Implementation
+
+We will create a new admission controller called PodSchedulerPolicy. It will be configured
+using a new API object called `PodPolicy`. We have chosen a generic name and API
+for this object because we want to use the same object for defining other policies based on pod
+characteristics, for example see #17097.
+
+PodSchedulerPolicy will iterate through the `Rules` of the `PodPolicy` and apply the
+first one that matches the pod. Each `Rule` is a string specifying the scheduler name
+to apply to the pod if the pod satisfies the rule's PolicyPredicate, which currently is
+just a wrapper over `PodSelector`.
+
+## API
+
+### PodPolicy
+
+```go
+// A PodPolicy is a list of rules that define the policy. Each rule is tested
+// in order, and the first one that matches is applied.
+type PodPolicy struct {
+	Rules []PodPolicyRule `json:"rules,omitempty"`
+}
+
+type PodPolicyRule struct {
+	// The rule defined by Policy is applied if the pod satisfies all of the criteria in PolicyPredicate
+	// (currently this just means satisfying the PodSelector in PolicyPredicate).
+	// The reason we use PolicyPredicate instead of just putting *PodSelector here is to allow
+	// us to cleanly add additional predicates in the future; they would be added to
+	// PolicyPredicate, allowing PodPolicyRule to stay simple.
+	PolicyPredicate *PolicyPredicate `json:"policyPredicate,omitempty"`
+	Policy string `json:"policy"`
```

Comment:
I think the YAML ends up as

````
kind: PodSchedulerPolicy
spec:
    policy:
        rules:
            - policyPredicate:
                  podSelector:
                      foo: bar
              policy: my-custom-scheduler
```

That's a WHOLE LOT of indent for a string.  It looks like you could kill one level of nesting with no loss of info or flexibility?

```
kind: PodSchedulerPolicy
spec:
    policy:
        - predicate:
              podSelector:
                  foo: bar
          value: my-custom-scheduler
```